### PR TITLE
Document all checkpoint policies in one place, on the JAX public API page.

### LINKED ISSUE #12417: Hard to find documentation of predefined jax checkpoint policies
### Description

I tried searching on https://jax.readthedocs.io/en/latest/search.html?q=checkpoint_dots for checkpoint_dots but couldn't find any documentation. The documentation of jax.checkpoint is helpful, but should provide more information on the predefined policies in jax.checkpoint_policies.

### What jax/jaxlib version are you using?

_No response_

### Which accelerator(s) are you using?

_No response_

### Additional System Info

_No response_

### PR DESCRIPTION
**Pull request by carlosgmartin**
Document all checkpoint policies (separately listed [here](https://docs.jax.dev/en/latest/notebooks/autodiff_remat.html#custom-policies-for-what-s-saveable) and [here](https://docs.jax.dev/en/latest/gradient-checkpointing.html#list-of-policies), currently) in one place: The JAX [public API page](file:///Users/carlos/Desktop/jax/docs/build/html/jax.html).

Fixes https://github.com/jax-ml/jax/issues/12417.

### CODE REVIEW

File: docs/notebooks/autodiff_remat.ipynb [hunk 1]
```diff
@@ -812,18 +812,6 @@
    "source": [
     "Another policy which refers to names is `jax.checkpoint_policies.save_only_these_names`.\n",
     "\n",
```
Comment (jakevdp) [inline docs/notebooks/autodiff_remat.ipynb:814]: Consider linking to the new doc here?

**Review by jakevdp — COMMENTED** [review]
Looks good, thanks! One comment below. Also requesting additional review of the checkpoint policy descriptions from @yashk2810

**Review by yashk2810 — APPROVED** [review]

File: docs/notebooks/autodiff_remat.ipynb (hunk 1 above)
Comment (carlosgmartin) [reply docs/notebooks/autodiff_remat.ipynb:814]: @jakevdp Updated.

File: docs/notebooks/autodiff_remat.md [hunk 2]
```diff
@@ -396,18 +396,6 @@ print_saved_residuals(loss_checkpoint2, params, x, y)
...
-* `nothing_saveable` (i.e. rematerialize everything, as if a custom policy were not being used at all)
-* `dots_saveable` or its alias `checkpoint_dots`
-* `dots_with_no_batch_dims_saveable` or its alias `checkpoint_dots_with_no_batch_dims`
-* `save_anything_but_these_names` (save any values except for the output of
-  `checkpoint_name` with any of the names given)
-* `save_any_names_but_these` (save only named values, i.e. any outputs of
-  `checkpoint_name`, except for those with the names given)
-* `save_only_these_names` (save only named values, and only among the names
-  given)
-
```
Comment (jakevdp) [inline docs/notebooks/autodiff_remat.md:410]: I think this needs to be synced (`pre-commit run jupytext --all-files` if you want to do so automatically)

**Comment by carlosgmartin** [comment]
@jakevdp @yashk2810 Any further changes you'd like me to make?
